toc/rfc/rfc-draft-enhance-stack-handling.md

```diff
@@ -0,0 +1,249 @@
+  anymore. Restarting and scaling of existing apps using the disabled stack is still possible.
+
+This requires the stack table to be extended by additional information
+regarding a stacks state as well as the endpoints to manage/list stacks
+as well. Additionally, the CLI and client libraries would need adoption
+to be able to present the newly added information to a CF user when
+calling "cf stacks".
+
+Following Datetimes MAY be saved per stack so this information CAN be
+used in consequent log lines, errors and deprecation notices and for the
+logic to influence stack usage:
+
+- deprecate_at -> Timestamp after which the deprecation notice will be
+  printed
+
+- lock_at -> Timestamp after which staging new apps with a that stack
+  fail
+
+- disable_at -> Timestamp after which restaging existing apps with a that
+  stack fail
+
+Stack lifecycle management is controlled exclusively by a set of timestamps, not by an explicit "state" field. The system determines the current state of a stack (active, deprecated, locked, or disabled) by comparing the current server time to these timestamps. This approach ensures that stack usability is automatically and unambiguously defined by time-based transitions. Additional information can already be added via the `description` field.
+
+- State transitions (deprecated_at → locked_at → disabled_at) are derived from the configured timestamps.
+- A validation check SHOULD be performed when setting the timestamps to ensure that they are in chronological order( deprecated_at → locked_at → disabled_at).
+- There is no explicit `state` field; the state is always computed at runtime from the timestamps.
+- For additional information the operator MAY reuse the existing `description` field in the stacks table to provide additional information about the stack.
+- In case a timestamp is not set, the stack is considered to be in the previous state indefinitely. This SHOULD also be considered in Error and Log messages.
+
+This model guarantees that stack usage is consistently enforced based on time, and all state transitions are predictable and transparent.
+It also provides a interface for an operator in which no externally timed api call for a stack state change is required.
+
+The CF API SHOULD use the provided timestamps to influence the logging and error messages in the following way:
+
+- It MAY add a log line into the Staging/Restaging logs of apps with a locked stack. It SHOULD produce the deprecation warning, optionally with color support, to underline the importance of a deprecated stack.
+- It MAY add the time since when a stack is deprecated/locked/disabled to the Staging/Restaging logs.
+- It MAY add the time when future state transitions will happen to the Staging/Restaging logs, so when is going to be locked/disabled.
+- The `description` field MAY be used in addition in Staging/Restaging logs to enable CF Admins to include a custom message in the apps staging logs
```

Comment:
CF CLI should also print out the current stack state and the deprecate/lock/disable_at timestamps.